library/src/text/raw.rs

```diff
@@ -127,75 +142,208 @@ impl Synthesize for RawElem {
 }
 
 impl Show for RawElem {
-    fn show(&self, _: &mut Vt, styles: StyleChain) -> SourceResult<Content> {
+    fn show(&self, vt: &mut Vt, styles: StyleChain) -> SourceResult<Content> {
         let text = self.text();
         let lang = self.lang(styles).as_ref().map(|s| s.to_lowercase());
+        let numbering = self.numbering(styles);
         let foreground = THEME
             .settings
             .foreground
             .map(to_typst)
             .map_or(Color::BLACK, Color::from)
             .into();
 
-        let mut realized = if matches!(lang.as_deref(), Some("typ" | "typst" | "typc")) {
+        let mut realized = if let Some(numbering) = numbering.as_ref() {
+            realize_with_numbering(text, lang, foreground, vt, styles, numbering)?
```

Comment:
I'm not a fan of having two completely separate code paths for this.

Reply:
Well, the logic of these two is very different, and I couldn't find a good way to merge them.